catalog/imaging/catalog-video-transformer/src/main/java/com/connexta/transformer/video/GeoBoxHandler.java

```diff
@@ -0,0 +1,137 @@
+
+    /**
+     * @param attributeName the name of the metacard attribute being generated
+     * @param latitude1     the name of the stanag 4609 field
+     * @param longitude1    the name of the stanag 4609 field
+     * @param latitude2     the name of the stanag 4609 field
+     * @param longitude2    the name of the stanag 4609 field
+     * @param latitude3     the name of the stanag 4609 field
+     * @param longitude3    the name of the stanag 4609 field
+     * @param latitude4     the name of the stanag 4609 field
+     * @param longitude4    the name of the stanag 4609 field
+     */
+    public GeoBoxHandler(String attributeName, String latitude1, String longitude1,
+            String latitude2, String longitude2, String latitude3, String longitude3,
+            String latitude4, String longitude4) {
+        super(attributeName);
+
+        this.latitude1 = latitude1;
+        this.longitude1 = longitude1;
+
+        this.latitude2 = latitude2;
+        this.longitude2 = longitude2;
+
+        this.latitude3 = latitude3;
+        this.longitude3 = longitude3;
+
+        this.latitude4 = latitude4;
+        this.longitude4 = longitude4;
+
+    }
+
+    @Override
+    public Optional<Attribute> asAttribute() {
+
+        int minimumListSize = map.values()
+                .stream()
+                .min((a, b) -> Integer.compare(a.size(), b.size()))
+                .orElse(Collections.emptyList())
+                .size();
+
+        List<String> pairs = new ArrayList<>();
+
+        for (int i = 0; i < minimumListSize; i++) {
+            pairs.add(String.format("POLYGON ((%f %f, %f %f, %f %f, %f %f, %f %f))",
+                    map.get(longitude1)
+                            .get(i),
+                    map.get(latitude1)
+                            .get(i),
+                    map.get(longitude2)
+                            .get(i),
+                    map.get(latitude2)
+                            .get(i),
+                    map.get(longitude3)
+                            .get(i),
+                    map.get(latitude3)
+                            .get(i),
+                    map.get(longitude4)
+                            .get(i),
+                    map.get(latitude4)
+                            .get(i),
+                    map.get(longitude1)
+                            .get(i),
+                    map.get(latitude1)
+                            .get(i)));
+        }
+
+        return asAttribute(pairs);
+    }
+
+    @Override
+    public void accept(KlvDataElement klvDataElement) {
+        if (!(klvDataElement instanceof KlvIntegerEncodedFloatingPoint)) {
+            LOGGER.warn(
+                    "non-KlvIntegerEncodedFloatingPoint data was passed to the GeoBoxHandler: name = {} klvDataElement = {}",
+                    klvDataElement.getName(),
+                    klvDataElement);
+            return;
+        }
+        map.putIfAbsent(klvDataElement.getName(), new LinkedList<>());
```

Comment:
Are you sure you want to use a `LinkedList` here? Inside the loop in `asAttribute()` you're calling `get(index)`, which will result in many iterations over the list elements.